src/platform/silabs/CHIPPlatformConfig.h

```diff
@@ -118,6 +118,8 @@
 #define CHIP_CONFIG_ICD_CLIENTS_SUPPORTED_PER_FABRIC SL_ICD_SUPPORTED_CLIENTS_PER_FABRIC
 #endif // CHIP_CONFIG_ICD_CLIENTS_SUPPORTED_PER_FABRIC
 
+#define CHIP_CONFIG_ICD_OBSERVERS_POOL_SIZE 3
```

Comment:
Suggestion so the Userconfig can still override this to increase it more
```suggestion
#ifndef CHIP_CONFIG_ICD_OBSERVERS_POOL_SIZE
#define CHIP_CONFIG_ICD_OBSERVERS_POOL_SIZE 3
#endif
```